src/web/app/pages-instructor/instructor-course-details-page/instructor-course-details-page.component.html

```diff
@@ -1,3 +1,143 @@
+    <div class="card-header text-white bg-primary font-weight-bold">Students</div>
+    <div class="card-body p-3">
+      <div class="card-header text-dark-blue bg-light-blue font-weight-bold">[{{courseDetails.course.id}}]</div>
+      <div class="card-body p-0">
+        <tm-student-list [courseId]="courseDetails.course.id" [sections]="sections" [useGrayHeading]="true" [enableRemindButton]="true">
+        </tm-student-list>
+      </div>
+    </div>
+  </ng-template>
+  <ng-template #remindStudentsModal let-modal>
+    <div class="modal-header alert-info">
+      <i class="fa fa-info-circle fa-2x"></i>
+      <h4 class="modal-title">Confirm sending join requests</h4>
+      <button type="button" class="close" aria-label="Close" (click)="modal.dismiss()">&times;</button>
+    </div>
+    <div class="modal-body">
+      Usually, there is no need to use this feature because TEAMMATES sends an automatic invite to students at the opening
+      time of each session. Send a join request to all yet-to-join students in {{courseDetails.course.id}} anyway?
+      <br>
+    </div>
+    <div class="modal-footer">
+      <button type="button" class="modal-btn-cancel btn-lg btn-default" (click)="modal.dismiss()">Cancel</button>
+      <button type="button" class="modal-btn-ok btn-lg btn-danger" (click)="remindAllStudentsFromCourse(courseDetails.course.id); modal.dismiss()">OK</button>
+    </div>
+  </ng-template>
+  <ng-template #deleteAllStudentsModal let-modal>
+    <div class="modal-header alert-danger">
+      <i class="fa fa-times-circle fa-2x"></i>
+      <h4 class="modal-title">Confirm Deletion</h4>
+      <button type="button" class="close" aria-label="Close" (click)="modal.dismiss()">&times;</button>
+    </div>
+    <div class="modal-body">
+      Are you sure you want to remove all students from the course {{courseDetails.course.id}}?
+      <br>
+    </div>
+    <div class="modal-footer">
+      <button type="button" class="modal-btn-cancel btn-lg btn-default" (click)="modal.dismiss()">Cancel</button>
+      <button type="button" class="modal-btn-ok btn-lg btn-danger" (click)="deleteAllStudentsFromCourse(courseDetails.course.id); modal.dismiss()">OK</button>
+    </div>
+  </ng-template>
+  <ng-template #studentTableModal let-modal>
+    <div class="modal-header">
+      <span>
+        Tips: <span class="btn-link" (click)="copyContent(courseStudentListAsCsv)">Click here</span>
+        to copy the table, then <kbd>Ctrl + V</kbd> or <kbd>⌘ + V</kbd> to PASTE to your Excel Workbook.
+      </span>
+      <button type="button" class="close" aria-label="Close" (click)="modal.dismiss('Cross click')">
+        <span aria-hidden="true">&times;</span>
+      </button>
+    </div>
+    <div class="modal-body">
+      <tm-ajax-preload *ngIf="loading"></tm-ajax-preload>
+      <div [innerHTML]="convertToHtmlTable(courseStudentListAsCsv)"></div>
+      <b>{{ isAjaxSuccess ? '' : 'Failed to load. Please reopen modal.' }}</b>
```

Comment:
Closing-then-reopening the modal when something fails does not sound very user friendly. As a user I would expect something inside the modal itself to be clickable to reload.